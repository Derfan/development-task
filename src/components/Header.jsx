import React from "react";
import styled from "styled-components";
import {withRouter} from "react-router-dom";

export const getCounter = (state, count) => {
    if (!state) return `Number of albums: ${count}`;

    const key = Object.keys(state)[0];

    switch (key) {
        case "photo":
            return `Photo ID: ${state[key].id}`;
        case "album":
            return `Number of photos: ${state[key].length}`;
        default:
            return ``
    }
};

export const Header = ({title, albumsCount, location: {state}}) => {
    const countText = getCounter(state, albumsCount);

    return (
        <HeaderBlock>
            <h1>{title}</h1>

            <p>{countText}</p>
        </HeaderBlock>
    );
};

const HeaderBlock = styled.header`
    display: flex;
    justify-content: space-between;
    align-items: center;
`;

export default withRouter(Header)
