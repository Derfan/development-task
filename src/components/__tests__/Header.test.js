import React from "react";
import {shallow} from "enzyme";
import {Header, getCounter} from "../Header";

const defaultProps = {
    title: "Test title",
    albumsCount: 5,
    location: {state: {}}
};

describe("Header", () => {
    const component = shallow(<Header {...defaultProps} />);

    it("render", () => {
        expect(component.find("h1")).toHaveLength(1);
        expect(component.find("p")).toHaveLength(1);
    });

    it("should return valid string", () => {
        const statePhoto = {photo: {id: 'test1'}};
        const stateAlbum = {album: [0, 1, 2, 3, 4]};

        expect(getCounter(statePhoto)).toEqual(`Photo ID: ${statePhoto.photo.id}`);
        expect(getCounter(stateAlbum)).toEqual(`Number of photos: ${stateAlbum.album.length}`);
    });
});
