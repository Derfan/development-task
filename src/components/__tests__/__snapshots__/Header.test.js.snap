// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    albumsCount={5}
    location={
      Object {
        "state": Object {},
      }
    }
    title="Test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1>
          Test title
        </h1>,
        <p>
          
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test title",
        },
        "ref": null,
        "rendered": "Test title",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    display: flex;
    justify-content: space-between;
    align-items: center;
",
        ],
      },
      "displayName": "styled.header",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "header",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1>
            Test title
          </h1>,
          <p>
            
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test title",
          },
          "ref": null,
          "rendered": "Test title",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    display: flex;
    justify-content: space-between;
    align-items: center;
",
          ],
        },
        "displayName": "styled.header",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "header",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
